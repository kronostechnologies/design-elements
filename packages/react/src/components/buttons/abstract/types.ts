import { ButtonHTMLAttributes } from 'react';
import { Size } from '../types';

export interface AbstractButtonProps extends ButtonHTMLAttributes<HTMLButtonElement> {
    focusable?: boolean;
    isMobile: boolean;
    size?: Size;
}
