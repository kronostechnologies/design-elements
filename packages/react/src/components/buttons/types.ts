import { FocusEventHandler, MouseEvent, KeyboardEvent } from 'react';
import { IconName } from '../icon/icon';
import { Size } from './abstract/types';

export type Type = 'submit' | 'button' | 'reset';

export type ButtonType =
    | 'primary'
    | 'secondary'
    | 'tertiary'
    | 'destructive-primary'
    | 'destructive-secondary'
    | 'destructive-tertiary';

export interface ButtonProps {
    id?: string;
    autofocus?: boolean;
    /**
     * Visual style
     * @default primary
     */
    buttonType: ButtonType;
    className?: string;
    disabled?: boolean;
    /**
     * @default true
     */
    focusable?: boolean;
    inverted?: boolean;
    label?: string;
    loading?: boolean;
    /**
     * @default Loading...
     */
    loadingLabel?: string;
    /**
     * Size variant
     * @default medium
     */
    size?: Size;
    title?: string;
    type?: Type;

    leftIconName?: IconName;
    rightIconName?: IconName;

    onClick?(event: MouseEvent<HTMLButtonElement>): void;
    onFocus?: FocusEventHandler<HTMLButtonElement>;
    onBlur?: FocusEventHandler<HTMLButtonElement>;
    onKeyDown?(event: KeyboardEvent<HTMLButtonElement>): void;
}
