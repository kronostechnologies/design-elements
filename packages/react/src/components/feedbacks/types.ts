import { DetailedHTMLProps, HTMLAttributes } from 'react';
import { FieldContainerProps } from '../field/types';

export type BaseSpanProps = Pick<DetailedHTMLProps<HTMLAttributes<HTMLSpanElement>, HTMLSpanElement>,
    | 'id' | 'className'
>;

type CommonFieldContainerProps = Pick<FieldContainerProps, 'noInvalidFieldIcon'>;

export interface InvalidFieldMessageProps extends BaseSpanProps, CommonFieldContainerProps {}
