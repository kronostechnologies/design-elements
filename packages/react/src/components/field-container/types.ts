import { TooltipProps } from '../tooltip/tooltip';

export interface FieldSlotIds {
    label?: string;
    hint?: string;
    invalid?: string;
}

export interface FieldControlProps {
    id?: string;
    ariaLabel?: string;
    ariaLabelledby?: string;
    ariaDescribedby?: string
    disabled?: boolean;
    required?: boolean;
    valid?: boolean;
    slotIds?: FieldSlotIds;
}

export interface FieldContainerProps extends FieldControlProps {
    fieldId?: string;
    className?: string;
    noMargin?: boolean;
    label?: string;
    hint?: string;
    validationErrorMessage: string;
    noInvalidFieldIcon?: boolean;
    tooltip?: TooltipProps;
}
