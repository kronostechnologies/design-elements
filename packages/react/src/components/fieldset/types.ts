import { FieldsetHTMLAttributes, HTMLAttributes, ReactNode } from 'react';

export type LegendSize = 'small' | 'medium' | 'large';

export interface LegendProps extends HTMLAttributes<HTMLLegendElement> {
    className?: string;
    size?: LegendSize;
    bold?: boolean;
    disabled?: boolean;
}

export type FieldsetOrientation = 'horizontal' | 'vertical';

type BaseFieldsetProps = Pick<FieldsetHTMLAttributes<HTMLFieldSetElement>,
    'disabled' | 'id' | 'aria-disabled' | 'aria-orientation' | 'aria-label'
>;

export interface FieldsetProps extends BaseFieldsetProps {
    legend?: LegendProps;
    orientation?: FieldsetOrientation;
    children: ReactNode;
}
