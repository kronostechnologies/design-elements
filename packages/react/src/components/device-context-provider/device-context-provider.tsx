import {
    createContext,
    FunctionComponent,
    PropsWithChildren,
    useCallback,
    useContext,
    useEffect,
    useRef,
    useState,
} from 'react';
import { breakpoints, Breakpoints } from '../../legacy-constants/breakpoints';

export type DeviceType = 'desktop' | 'tablet' | 'mobile';

export interface DeviceContextProviderProps {
    staticDevice?: DeviceType;
}

export interface DeviceContextProps {
    device: DeviceType;
    isDesktop: boolean;
    isTablet: boolean;
    isMobile: boolean;
    breakpoints: Breakpoints;
}

function getDeviceContext(deviceName: DeviceType): DeviceContextProps {
    return {
        device: deviceName,
        isDesktop: deviceName === 'desktop',
        isTablet: deviceName === 'tablet',
        isMobile: deviceName === 'mobile',
        breakpoints,
    };
}

function getDeviceType(): DeviceType {
    let currentDevice: DeviceType = 'desktop';
    const screenWidth = (window.innerWidth || document.documentElement.clientWidth);

    if (screenWidth >= breakpoints.desktop) {
        currentDevice = 'desktop';
    } else if (screenWidth < breakpoints.desktop && screenWidth > breakpoints.mobile) {
        currentDevice = 'tablet';
    } else if (screenWidth <= breakpoints.mobile) {
        currentDevice = 'mobile';
    }

    return currentDevice;
}

const DeviceContext = createContext<DeviceContextProps>(getDeviceContext(getDeviceType()));

export const DeviceContextProvider: FunctionComponent<PropsWithChildren<DeviceContextProviderProps>> = ({
    children,
    staticDevice,
}) => {
    const deviceType = staticDevice || getDeviceType();

    const previousDeviceType = useRef<DeviceType>(deviceType);
    const previousStaticDevice = useRef<DeviceType | undefined>(staticDevice);
    const [deviceContext, setDeviceContext] = useState<DeviceContextProps>(getDeviceContext(deviceType));

    if (staticDevice !== previousStaticDevice.current) {
        previousStaticDevice.current = staticDevice;
        setDeviceContext(getDeviceContext(deviceType));
    }

    const handleScreenResize = useCallback(() => {
        const currentDeviceType = getDeviceType();

        if (currentDeviceType !== previousDeviceType.current) {
            previousDeviceType.current = currentDeviceType;
            setDeviceContext(getDeviceContext(currentDeviceType));
        }
    }, []);

    useEffect(() => {
        if (!staticDevice) {
            window.addEventListener('resize', handleScreenResize);
            return () => {
                window.removeEventListener('resize', handleScreenResize);
            };
        }
        return undefined;
    }, [handleScreenResize, staticDevice]);

    return (
        <DeviceContext.Provider value={deviceContext}>
            {children}
        </DeviceContext.Provider>
    );
};

export function useDeviceContext(): DeviceContextProps {
    return useContext(DeviceContext);
}
