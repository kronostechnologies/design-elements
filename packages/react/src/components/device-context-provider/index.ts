export { DeviceContextProvider, useDeviceContext } from './device-context-provider';
export type { DeviceContextProviderProps, DeviceType } from './device-context-provider';
