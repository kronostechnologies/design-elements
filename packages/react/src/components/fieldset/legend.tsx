import { type FC, HTMLAttributes } from 'react';
import styled from 'styled-components';
import { useDeviceContext } from '../device-context-provider';

export type LegendSize = 'small' | 'medium' | 'large';

export interface LegendProps extends HTMLAttributes<HTMLLegendElement> {
    className?: string;
    size?: LegendSize;
    bold?: boolean;
}

function getFontSize({ $isMobile, $size = 'medium' }: { $isMobile: boolean, $size?: LegendProps['size'] }): string {
    const mobileSizes = { small: '0.75rem', medium: '0.875rem', large: '1rem' };
    const desktopSizes = { small: '0.625rem', medium: '0.75rem', large: '0.875rem' };
    const sizes = $isMobile ? mobileSizes : desktopSizes;
    return sizes[$size] || sizes.medium;
}

export const StyledLegend = styled.legend<{
    $size: LegendProps['size'],
    $bold: LegendProps['bold'],
    $isMobile: boolean,
}>`
    color: ${({ theme }) => theme.component['legend-text-color']};
    font-size: ${getFontSize};
    font-weight: ${({ $bold }) => ($bold ? 'var(--font-bold)' : 'var(--font-normal)')};
    margin: 0 0 var(--spacing-1x);
`;

export const Legend: FC<LegendProps> = ({
    children,
    size,
    bold,
    ...props
}) => {
    const { isMobile } = useDeviceContext();

    return (
        <StyledLegend
            datatest-id="legend"
            $size={size}
            $bold={bold}
            $isMobile={isMobile}
            {...props /* eslint-disable-line react/jsx-props-no-spreading */}
        >
            {children}
        </StyledLegend>
    );
};
