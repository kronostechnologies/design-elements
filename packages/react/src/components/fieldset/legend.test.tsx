import { mountWithProviders } from '../../test-utils/renderer';
import { DeviceContextProviderProps } from '../device-context-provider/device-context-provider';
import { Legend } from './legend';
import { LegendSize } from './types';

describe('Legend Component', () => {
    const sizes: LegendSize[] = ['xxsmall', 'xsmall', 'small', 'medium', 'large'];
    const boldStates = [true, false];
    const devices: DeviceContextProviderProps['staticDevice'][] = ['mobile', 'desktop'];

    describe('Styling', () => {
        describe.each(devices)('Device: %s', (device) => {
            describe.each(sizes)('Size: %s', (size) => {
                it.each(boldStates)('Bold: %s', (bold) => {
                    const tree = mountWithProviders(
                        <Legend size={size} bold={bold}>Sample Legend</Legend>,
                        { wrappingComponentProps: { staticDevice: device } },
                    );
                    expect(tree).toMatchSnapshot();
                });
            });
        });
    });
});
