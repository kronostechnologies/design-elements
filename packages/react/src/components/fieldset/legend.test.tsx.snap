// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Legend Component Device: desktop renders with all combinations of props 1`] = `
.c0 {
  display: table;
  color: #B7BBC2;
  font-size: 0.625rem;
  font-weight: var(--font-bold);
  margin: 0 0 var(--spacing-1x);
}

<Legend
  bold={true}
  disabled={true}
  size="small"
>
  <styled.legend
    bold={true}
    datatest-id="legend"
    disabled={true}
    isMobile={false}
    size="small"
  >
    <legend
      className="c0"
      disabled={true}
      size="small"
    >
      Sample Legend
    </legend>
  </styled.legend>
</Legend>
`;

exports[`Legend Component Device: desktop renders with all combinations of props 2`] = `
.c0 {
  display: table;
  color: #1B1C1E;
  font-size: 0.625rem;
  font-weight: var(--font-bold);
  margin: 0 0 var(--spacing-1x);
}

<Legend
  bold={true}
  disabled={false}
  size="small"
>
  <styled.legend
    bold={true}
    datatest-id="legend"
    disabled={false}
    isMobile={false}
    size="small"
  >
    <legend
      className="c0"
      disabled={false}
      size="small"
    >
      Sample Legend
    </legend>
  </styled.legend>
</Legend>
`;

exports[`Legend Component Device: desktop renders with all combinations of props 3`] = `
.c0 {
  display: table;
  color: #B7BBC2;
  font-size: 0.625rem;
  font-weight: var(--font-normal);
  margin: 0 0 var(--spacing-1x);
}

<Legend
  bold={false}
  disabled={true}
  size="small"
>
  <styled.legend
    bold={false}
    datatest-id="legend"
    disabled={true}
    isMobile={false}
    size="small"
  >
    <legend
      className="c0"
      disabled={true}
      size="small"
    >
      Sample Legend
    </legend>
  </styled.legend>
</Legend>
`;

exports[`Legend Component Device: desktop renders with all combinations of props 4`] = `
.c0 {
  display: table;
  color: #1B1C1E;
  font-size: 0.625rem;
  font-weight: var(--font-normal);
  margin: 0 0 var(--spacing-1x);
}

<Legend
  bold={false}
  disabled={false}
  size="small"
>
  <styled.legend
    bold={false}
    datatest-id="legend"
    disabled={false}
    isMobile={false}
    size="small"
  >
    <legend
      className="c0"
      disabled={false}
      size="small"
    >
      Sample Legend
    </legend>
  </styled.legend>
</Legend>
`;

exports[`Legend Component Device: desktop renders with all combinations of props 5`] = `
.c0 {
  display: table;
  color: #B7BBC2;
  font-size: 0.75rem;
  font-weight: var(--font-bold);
  margin: 0 0 var(--spacing-1x);
}

<Legend
  bold={true}
  disabled={true}
  size="medium"
>
  <styled.legend
    bold={true}
    datatest-id="legend"
    disabled={true}
    isMobile={false}
    size="medium"
  >
    <legend
      className="c0"
      disabled={true}
      size="medium"
    >
      Sample Legend
    </legend>
  </styled.legend>
</Legend>
`;

exports[`Legend Component Device: desktop renders with all combinations of props 6`] = `
.c0 {
  display: table;
  color: #1B1C1E;
  font-size: 0.75rem;
  font-weight: var(--font-bold);
  margin: 0 0 var(--spacing-1x);
}

<Legend
  bold={true}
  disabled={false}
  size="medium"
>
  <styled.legend
    bold={true}
    datatest-id="legend"
    disabled={false}
    isMobile={false}
    size="medium"
  >
    <legend
      className="c0"
      disabled={false}
      size="medium"
    >
      Sample Legend
    </legend>
  </styled.legend>
</Legend>
`;

exports[`Legend Component Device: desktop renders with all combinations of props 7`] = `
.c0 {
  display: table;
  color: #B7BBC2;
  font-size: 0.75rem;
  font-weight: var(--font-normal);
  margin: 0 0 var(--spacing-1x);
}

<Legend
  bold={false}
  disabled={true}
  size="medium"
>
  <styled.legend
    bold={false}
    datatest-id="legend"
    disabled={true}
    isMobile={false}
    size="medium"
  >
    <legend
      className="c0"
      disabled={true}
      size="medium"
    >
      Sample Legend
    </legend>
  </styled.legend>
</Legend>
`;

exports[`Legend Component Device: desktop renders with all combinations of props 8`] = `
.c0 {
  display: table;
  color: #1B1C1E;
  font-size: 0.75rem;
  font-weight: var(--font-normal);
  margin: 0 0 var(--spacing-1x);
}

<Legend
  bold={false}
  disabled={false}
  size="medium"
>
  <styled.legend
    bold={false}
    datatest-id="legend"
    disabled={false}
    isMobile={false}
    size="medium"
  >
    <legend
      className="c0"
      disabled={false}
      size="medium"
    >
      Sample Legend
    </legend>
  </styled.legend>
</Legend>
`;

exports[`Legend Component Device: desktop renders with all combinations of props 9`] = `
.c0 {
  display: table;
  color: #B7BBC2;
  font-size: 0.875rem;
  font-weight: var(--font-bold);
  margin: 0 0 var(--spacing-1x);
}

<Legend
  bold={true}
  disabled={true}
  size="large"
>
  <styled.legend
    bold={true}
    datatest-id="legend"
    disabled={true}
    isMobile={false}
    size="large"
  >
    <legend
      className="c0"
      disabled={true}
      size="large"
    >
      Sample Legend
    </legend>
  </styled.legend>
</Legend>
`;

exports[`Legend Component Device: desktop renders with all combinations of props 10`] = `
.c0 {
  display: table;
  color: #1B1C1E;
  font-size: 0.875rem;
  font-weight: var(--font-bold);
  margin: 0 0 var(--spacing-1x);
}

<Legend
  bold={true}
  disabled={false}
  size="large"
>
  <styled.legend
    bold={true}
    datatest-id="legend"
    disabled={false}
    isMobile={false}
    size="large"
  >
    <legend
      className="c0"
      disabled={false}
      size="large"
    >
      Sample Legend
    </legend>
  </styled.legend>
</Legend>
`;

exports[`Legend Component Device: desktop renders with all combinations of props 11`] = `
.c0 {
  display: table;
  color: #B7BBC2;
  font-size: 0.875rem;
  font-weight: var(--font-normal);
  margin: 0 0 var(--spacing-1x);
}

<Legend
  bold={false}
  disabled={true}
  size="large"
>
  <styled.legend
    bold={false}
    datatest-id="legend"
    disabled={true}
    isMobile={false}
    size="large"
  >
    <legend
      className="c0"
      disabled={true}
      size="large"
    >
      Sample Legend
    </legend>
  </styled.legend>
</Legend>
`;

exports[`Legend Component Device: desktop renders with all combinations of props 12`] = `
.c0 {
  display: table;
  color: #1B1C1E;
  font-size: 0.875rem;
  font-weight: var(--font-normal);
  margin: 0 0 var(--spacing-1x);
}

<Legend
  bold={false}
  disabled={false}
  size="large"
>
  <styled.legend
    bold={false}
    datatest-id="legend"
    disabled={false}
    isMobile={false}
    size="large"
  >
    <legend
      className="c0"
      disabled={false}
      size="large"
    >
      Sample Legend
    </legend>
  </styled.legend>
</Legend>
`;

exports[`Legend Component Device: desktop renders with bold false 1`] = `
.c0 {
  display: table;
  color: #1B1C1E;
  font-size: 0.75rem;
  font-weight: var(--font-normal);
  margin: 0 0 var(--spacing-1x);
}

<Legend
  bold={false}
>
  <styled.legend
    bold={false}
    datatest-id="legend"
    isMobile={false}
  >
    <legend
      className="c0"
    >
      Sample Legend
    </legend>
  </styled.legend>
</Legend>
`;

exports[`Legend Component Device: desktop renders with bold true 1`] = `
.c0 {
  display: table;
  color: #1B1C1E;
  font-size: 0.75rem;
  font-weight: var(--font-bold);
  margin: 0 0 var(--spacing-1x);
}

<Legend
  bold={true}
>
  <styled.legend
    bold={true}
    datatest-id="legend"
    isMobile={false}
  >
    <legend
      className="c0"
    >
      Sample Legend
    </legend>
  </styled.legend>
</Legend>
`;

exports[`Legend Component Device: desktop renders with disabled false 1`] = `
.c0 {
  display: table;
  color: #1B1C1E;
  font-size: 0.75rem;
  font-weight: var(--font-normal);
  margin: 0 0 var(--spacing-1x);
}

<Legend
  disabled={false}
>
  <styled.legend
    datatest-id="legend"
    disabled={false}
    isMobile={false}
  >
    <legend
      className="c0"
      disabled={false}
    >
      Sample Legend
    </legend>
  </styled.legend>
</Legend>
`;

exports[`Legend Component Device: desktop renders with disabled true 1`] = `
.c0 {
  display: table;
  color: #B7BBC2;
  font-size: 0.75rem;
  font-weight: var(--font-normal);
  margin: 0 0 var(--spacing-1x);
}

<Legend
  disabled={true}
>
  <styled.legend
    datatest-id="legend"
    disabled={true}
    isMobile={false}
  >
    <legend
      className="c0"
      disabled={true}
    >
      Sample Legend
    </legend>
  </styled.legend>
</Legend>
`;

exports[`Legend Component Device: desktop renders with size large 1`] = `
.c0 {
  display: table;
  color: #1B1C1E;
  font-size: 0.875rem;
  font-weight: var(--font-normal);
  margin: 0 0 var(--spacing-1x);
}

<Legend
  size="large"
>
  <styled.legend
    datatest-id="legend"
    isMobile={false}
    size="large"
  >
    <legend
      className="c0"
      size="large"
    >
      Sample Legend
    </legend>
  </styled.legend>
</Legend>
`;

exports[`Legend Component Device: desktop renders with size medium 1`] = `
.c0 {
  display: table;
  color: #1B1C1E;
  font-size: 0.75rem;
  font-weight: var(--font-normal);
  margin: 0 0 var(--spacing-1x);
}

<Legend
  size="medium"
>
  <styled.legend
    datatest-id="legend"
    isMobile={false}
    size="medium"
  >
    <legend
      className="c0"
      size="medium"
    >
      Sample Legend
    </legend>
  </styled.legend>
</Legend>
`;

exports[`Legend Component Device: desktop renders with size small 1`] = `
.c0 {
  display: table;
  color: #1B1C1E;
  font-size: 0.625rem;
  font-weight: var(--font-normal);
  margin: 0 0 var(--spacing-1x);
}

<Legend
  size="small"
>
  <styled.legend
    datatest-id="legend"
    isMobile={false}
    size="small"
  >
    <legend
      className="c0"
      size="small"
    >
      Sample Legend
    </legend>
  </styled.legend>
</Legend>
`;

exports[`Legend Component Device: mobile renders with all combinations of props 1`] = `
.c0 {
  display: table;
  color: #B7BBC2;
  font-size: 0.75rem;
  font-weight: var(--font-bold);
  margin: 0 0 var(--spacing-1x);
}

<Legend
  bold={true}
  disabled={true}
  size="small"
>
  <styled.legend
    bold={true}
    datatest-id="legend"
    disabled={true}
    isMobile={true}
    size="small"
  >
    <legend
      className="c0"
      disabled={true}
      size="small"
    >
      Sample Legend
    </legend>
  </styled.legend>
</Legend>
`;

exports[`Legend Component Device: mobile renders with all combinations of props 2`] = `
.c0 {
  display: table;
  color: #1B1C1E;
  font-size: 0.75rem;
  font-weight: var(--font-bold);
  margin: 0 0 var(--spacing-1x);
}

<Legend
  bold={true}
  disabled={false}
  size="small"
>
  <styled.legend
    bold={true}
    datatest-id="legend"
    disabled={false}
    isMobile={true}
    size="small"
  >
    <legend
      className="c0"
      disabled={false}
      size="small"
    >
      Sample Legend
    </legend>
  </styled.legend>
</Legend>
`;

exports[`Legend Component Device: mobile renders with all combinations of props 3`] = `
.c0 {
  display: table;
  color: #B7BBC2;
  font-size: 0.75rem;
  font-weight: var(--font-normal);
  margin: 0 0 var(--spacing-1x);
}

<Legend
  bold={false}
  disabled={true}
  size="small"
>
  <styled.legend
    bold={false}
    datatest-id="legend"
    disabled={true}
    isMobile={true}
    size="small"
  >
    <legend
      className="c0"
      disabled={true}
      size="small"
    >
      Sample Legend
    </legend>
  </styled.legend>
</Legend>
`;

exports[`Legend Component Device: mobile renders with all combinations of props 4`] = `
.c0 {
  display: table;
  color: #1B1C1E;
  font-size: 0.75rem;
  font-weight: var(--font-normal);
  margin: 0 0 var(--spacing-1x);
}

<Legend
  bold={false}
  disabled={false}
  size="small"
>
  <styled.legend
    bold={false}
    datatest-id="legend"
    disabled={false}
    isMobile={true}
    size="small"
  >
    <legend
      className="c0"
      disabled={false}
      size="small"
    >
      Sample Legend
    </legend>
  </styled.legend>
</Legend>
`;

exports[`Legend Component Device: mobile renders with all combinations of props 5`] = `
.c0 {
  display: table;
  color: #B7BBC2;
  font-size: 0.875rem;
  font-weight: var(--font-bold);
  margin: 0 0 var(--spacing-1x);
}

<Legend
  bold={true}
  disabled={true}
  size="medium"
>
  <styled.legend
    bold={true}
    datatest-id="legend"
    disabled={true}
    isMobile={true}
    size="medium"
  >
    <legend
      className="c0"
      disabled={true}
      size="medium"
    >
      Sample Legend
    </legend>
  </styled.legend>
</Legend>
`;

exports[`Legend Component Device: mobile renders with all combinations of props 6`] = `
.c0 {
  display: table;
  color: #1B1C1E;
  font-size: 0.875rem;
  font-weight: var(--font-bold);
  margin: 0 0 var(--spacing-1x);
}

<Legend
  bold={true}
  disabled={false}
  size="medium"
>
  <styled.legend
    bold={true}
    datatest-id="legend"
    disabled={false}
    isMobile={true}
    size="medium"
  >
    <legend
      className="c0"
      disabled={false}
      size="medium"
    >
      Sample Legend
    </legend>
  </styled.legend>
</Legend>
`;

exports[`Legend Component Device: mobile renders with all combinations of props 7`] = `
.c0 {
  display: table;
  color: #B7BBC2;
  font-size: 0.875rem;
  font-weight: var(--font-normal);
  margin: 0 0 var(--spacing-1x);
}

<Legend
  bold={false}
  disabled={true}
  size="medium"
>
  <styled.legend
    bold={false}
    datatest-id="legend"
    disabled={true}
    isMobile={true}
    size="medium"
  >
    <legend
      className="c0"
      disabled={true}
      size="medium"
    >
      Sample Legend
    </legend>
  </styled.legend>
</Legend>
`;

exports[`Legend Component Device: mobile renders with all combinations of props 8`] = `
.c0 {
  display: table;
  color: #1B1C1E;
  font-size: 0.875rem;
  font-weight: var(--font-normal);
  margin: 0 0 var(--spacing-1x);
}

<Legend
  bold={false}
  disabled={false}
  size="medium"
>
  <styled.legend
    bold={false}
    datatest-id="legend"
    disabled={false}
    isMobile={true}
    size="medium"
  >
    <legend
      className="c0"
      disabled={false}
      size="medium"
    >
      Sample Legend
    </legend>
  </styled.legend>
</Legend>
`;

exports[`Legend Component Device: mobile renders with all combinations of props 9`] = `
.c0 {
  display: table;
  color: #B7BBC2;
  font-size: 1rem;
  font-weight: var(--font-bold);
  margin: 0 0 var(--spacing-1x);
}

<Legend
  bold={true}
  disabled={true}
  size="large"
>
  <styled.legend
    bold={true}
    datatest-id="legend"
    disabled={true}
    isMobile={true}
    size="large"
  >
    <legend
      className="c0"
      disabled={true}
      size="large"
    >
      Sample Legend
    </legend>
  </styled.legend>
</Legend>
`;

exports[`Legend Component Device: mobile renders with all combinations of props 10`] = `
.c0 {
  display: table;
  color: #1B1C1E;
  font-size: 1rem;
  font-weight: var(--font-bold);
  margin: 0 0 var(--spacing-1x);
}

<Legend
  bold={true}
  disabled={false}
  size="large"
>
  <styled.legend
    bold={true}
    datatest-id="legend"
    disabled={false}
    isMobile={true}
    size="large"
  >
    <legend
      className="c0"
      disabled={false}
      size="large"
    >
      Sample Legend
    </legend>
  </styled.legend>
</Legend>
`;

exports[`Legend Component Device: mobile renders with all combinations of props 11`] = `
.c0 {
  display: table;
  color: #B7BBC2;
  font-size: 1rem;
  font-weight: var(--font-normal);
  margin: 0 0 var(--spacing-1x);
}

<Legend
  bold={false}
  disabled={true}
  size="large"
>
  <styled.legend
    bold={false}
    datatest-id="legend"
    disabled={true}
    isMobile={true}
    size="large"
  >
    <legend
      className="c0"
      disabled={true}
      size="large"
    >
      Sample Legend
    </legend>
  </styled.legend>
</Legend>
`;

exports[`Legend Component Device: mobile renders with all combinations of props 12`] = `
.c0 {
  display: table;
  color: #1B1C1E;
  font-size: 1rem;
  font-weight: var(--font-normal);
  margin: 0 0 var(--spacing-1x);
}

<Legend
  bold={false}
  disabled={false}
  size="large"
>
  <styled.legend
    bold={false}
    datatest-id="legend"
    disabled={false}
    isMobile={true}
    size="large"
  >
    <legend
      className="c0"
      disabled={false}
      size="large"
    >
      Sample Legend
    </legend>
  </styled.legend>
</Legend>
`;

exports[`Legend Component Device: mobile renders with bold false 1`] = `
.c0 {
  display: table;
  color: #1B1C1E;
  font-size: 0.875rem;
  font-weight: var(--font-normal);
  margin: 0 0 var(--spacing-1x);
}

<Legend
  bold={false}
>
  <styled.legend
    bold={false}
    datatest-id="legend"
    isMobile={true}
  >
    <legend
      className="c0"
    >
      Sample Legend
    </legend>
  </styled.legend>
</Legend>
`;

exports[`Legend Component Device: mobile renders with bold true 1`] = `
.c0 {
  display: table;
  color: #1B1C1E;
  font-size: 0.875rem;
  font-weight: var(--font-bold);
  margin: 0 0 var(--spacing-1x);
}

<Legend
  bold={true}
>
  <styled.legend
    bold={true}
    datatest-id="legend"
    isMobile={true}
  >
    <legend
      className="c0"
    >
      Sample Legend
    </legend>
  </styled.legend>
</Legend>
`;

exports[`Legend Component Device: mobile renders with disabled false 1`] = `
.c0 {
  display: table;
  color: #1B1C1E;
  font-size: 0.875rem;
  font-weight: var(--font-normal);
  margin: 0 0 var(--spacing-1x);
}

<Legend
  disabled={false}
>
  <styled.legend
    datatest-id="legend"
    disabled={false}
    isMobile={true}
  >
    <legend
      className="c0"
      disabled={false}
    >
      Sample Legend
    </legend>
  </styled.legend>
</Legend>
`;

exports[`Legend Component Device: mobile renders with disabled true 1`] = `
.c0 {
  display: table;
  color: #B7BBC2;
  font-size: 0.875rem;
  font-weight: var(--font-normal);
  margin: 0 0 var(--spacing-1x);
}

<Legend
  disabled={true}
>
  <styled.legend
    datatest-id="legend"
    disabled={true}
    isMobile={true}
  >
    <legend
      className="c0"
      disabled={true}
    >
      Sample Legend
    </legend>
  </styled.legend>
</Legend>
`;

exports[`Legend Component Device: mobile renders with size large 1`] = `
.c0 {
  display: table;
  color: #1B1C1E;
  font-size: 1rem;
  font-weight: var(--font-normal);
  margin: 0 0 var(--spacing-1x);
}

<Legend
  size="large"
>
  <styled.legend
    datatest-id="legend"
    isMobile={true}
    size="large"
  >
    <legend
      className="c0"
      size="large"
    >
      Sample Legend
    </legend>
  </styled.legend>
</Legend>
`;

exports[`Legend Component Device: mobile renders with size medium 1`] = `
.c0 {
  display: table;
  color: #1B1C1E;
  font-size: 0.875rem;
  font-weight: var(--font-normal);
  margin: 0 0 var(--spacing-1x);
}

<Legend
  size="medium"
>
  <styled.legend
    datatest-id="legend"
    isMobile={true}
    size="medium"
  >
    <legend
      className="c0"
      size="medium"
    >
      Sample Legend
    </legend>
  </styled.legend>
</Legend>
`;

exports[`Legend Component Device: mobile renders with size small 1`] = `
.c0 {
  display: table;
  color: #1B1C1E;
  font-size: 0.75rem;
  font-weight: var(--font-normal);
  margin: 0 0 var(--spacing-1x);
}

<Legend
  size="small"
>
  <styled.legend
    datatest-id="legend"
    isMobile={true}
    size="small"
  >
    <legend
      className="c0"
      size="small"
    >
      Sample Legend
    </legend>
  </styled.legend>
</Legend>
`;
